js/ui/map.js

```diff
@@ -734,6 +734,23 @@ class Map extends Camera {
     }
 
     /**
+     * Returns a Boolean indicating whether the source is loaded.
+     *
+     * @param {string} id The ID of the source to be checked.
+     * @returns {boolean} A Boolean indicating whether the source is loaded.
+     */
+    isSourceLoaded(id) {
+        const source = this.style.sourceCaches[id];
+        if (source === undefined) {
+            this.fire('error', {
+                error: new Error(`There is no source with ID '${id}'`)
+            });
+            return;
+        }
```

Comment:
While it's true that the contributing guidelines [say to use error events](https://github.com/mapbox/mapbox-gl-js/blob/81d6715f8cf13c87ab930257a86440fbd0f3a91f/CONTRIBUTING.md#L98), I think `throw`ing an error might be more appropriate here, because we know immediately(synchronously) that there's an error. 

When, as in this case, synchronous error reporting is possible, I think it's preferable, as it's generally easier to diagnose/fix.  (E.g.: because of the `return;`, a user of this method might not even notice the error, since something like `if (map.isSourceLoaded(id)) { ... }` would still work without blowing up.)